playbooks/openstack/openshift-cluster/launch.yml

```diff
@@ -128,6 +128,6 @@
       - parsed_outputs.node_floating_ips
       - parsed_outputs.infra_floating_ips
 
-- include: update.yml
+- include: setup.yml
```

Comment:
I have the feeling that current playbooks have the following layout:
* `update.yml` contains the idempotent tasks that can be run both at cluster creation and at cluster update;
* `launch.yml` contains the non-idempotent tasks that must be run only at cluster creation.

For ex., each time `aws/…/launch.yml` is invoked, new VMs will be spawned. That’s why the `ec2` tasks have been isolated in `launch.yml` and not in `update.yml`.

If that’s true, it’s worth noticing that for the OpenStack cluster, the IaaS resources are managed by a HEAT template that can provide idempotency.

If we use
```
if the stack doesn’t exist then
  create the stack
else
  update the stack
```
then we have something which is idempotent.

The above pattern is already something we have in `openstack/…/launch.yml`

I have the feeling that what should be done is:
* to move from `openstack/…/launch.yml` to `openstack/…/update.yml` everything that is idempotent, that can be harmlessly relaunched. This includes… everything, the HEAT stack creation/update, the ssh checks, etc.
* to keep inside `openstack/…/launch.yml` only the `include: update.yml` and `include: list.yml` statements.

This would make `update` and `create` do exactly the same thing. This might look weird, but it is definitively in the spirit of Ansible.